mvc/overview.rst

```diff
@@ -1,8 +1,224 @@
-.. include:: /../common/stub-topic.txt
+Overview of ASP.NET MVC
+=======================
+By `Tom Archer`_, `Daniel Roth`_
 
-|stub-icon| Overview of ASP.NET MVC
-===================================
+.. image:: overview/_static/mvc.png
+  :align: right
 
-.. include:: /../common/stub-notice.txt
+ASP.NET MVC is a lightweight, cross-platform, highly testable Web application development framework that separates an application into three main components: model, view, and controller:
 
-.. _issue: https://github.com/aspnet/Docs/issues/113
+- :doc:`Models </models/index>` are the parts of the application that implement the application's state. Often model objects retrieve and store model state in a database. For example, a Product object might retrieve information from a database, operate on it, and then write updated information back to a Products table in a SQL Server database.
+
+- :doc:`Views </views/index>` are responsible for displaying the application's user interface (UI). Typically, this UI is created from the model data. An example would be an edit view of a Products table that displays text boxes, drop-down lists, and check boxes based on the current state of the Product object being represented in the view.
+
+- :doc:`Controllers </controllers/index>` process incoming requests, work with the model, and ultimately return a view to the client that displays the user interface.
+
+This delineation of responsibilities helps you scale the application in terms of complexity because it’s much easier to code, debug, and test something (model, view, or controller) that has a single job rather than something that performs multiple tasks.
+
+Features
+--------
+
+ASP.NET MVC provides the following features:
+  - `Routing`_
+  - `Model binding`_
+  - `Request validation`_
+  - `Razor view engine`_
+  - `Strongly typed views`_
+  - `HTML Helpers`_
+  - `Tag Helpers`_
+  - `View Components`_
+  - `Dependency Injection`_
+  - `Filters`_
+  - `Areas`_
+  - `Mobile development`_
+  - `Web UI and Web APIs`_
+  - `Test-driven Development`_
+  - `Built on top of ASP.NET 5`_
+  - `Tooling`_
+
+Routing
+-------
+
+MVC is built on :doc:`ASP.NET Routing </controllers/routing>`, a powerful URL-mapping component that lets you build applications that have comprehensible and searchable URLs. This enables you to define your application's URL naming patterns that work well for search engine optimization (SEO) and for link generation. You can define your routes using a convenient route template syntax that supports route value constraints, defaults and optional values.
+
+*Convention-based routing* enables you to globally define the URL formats that your application accepts and how each of those formats maps to a specific action method on given controller. When an incoming request is received, the routing engine parses the incoming URL and matches it to one of the defined URL formats, and calls the associated controller's action method. 
```

Comment:
I think we would usually say a "route template" or "route pattern" but never "route format." The term "format" is used in the preceding paragraph.